plugins/woocommerce/src/Internal/Utilities/URL.php

```diff
@@ -124,8 +131,11 @@ private function preprocess() {
 	 * without touching the filesystem.
 	 */
 	private function process_path() {
-		$segments          = explode( '/', $this->components['path'] );
-		$this->is_absolute = substr( $this->components['path'], 0, 1 ) === '/';
+		$segments           = explode( '/', $this->components['path'] );
+		$this->is_absolute  = substr( $this->components['path'], 0, 1 ) === '/' || ! empty( $this->components['host'] );
```

Comment:
does `absolute` here mean pointing to a specific file (as opposed to a relative URL or file)? if so, then what would be the case where `! empty( $this->components['host'] )` would also be needed (just curious)?

Reply:
Yes, in the case of file URLs we are checking if it represents an absolute filepath, whether that is a directory or a file. 

For URLs with host components (generally, anything that is not a file URL), we also want to treat them as being absolute specifically in relation to how any leading directory traversals are handled.

The inputs and outputs we want (simplified, but covers this aspect):

| Type | Unprocessed (Host+)Path | Processed Path |
| --- | --- | --- |
| **Absolute** | `/../../remainder` | `/remainder` |
| **Absolute** | `example.com/../../remainder` | `example.com/remainder` |
| **Relative** | `../../remainder` | `../../remainder` |